src/io/flutter/inspector/InspectorService.java

```diff
@@ -121,6 +116,55 @@ public void addClient(InspectorServiceClient client) {
     return instanceRefFuture.thenComposeAsync(this::parseDiagnosticsNode);
   }
 
+  /**
+   * Returns a CompletableFuture with a Map of property names to Observatory
+   * InstanceRef objects.
+   *
+   * It would be nice if the Observatory protocol provided a built in method
+   * to get InstanceRef objects for a list of properties but this is
+   * sufficient although slightly less efficient. The Observatory protocol
+   * does provide fast access to all fields as part of an Instance object
+   * but that is inadequate as for many Flutter data objects that we want
+   * to display visually we care about properties that are not neccesarily
+   * fields.
+   *
+   * The future will immediately complete to null if the inspectorInstanceRef is null.
+   */
+  public CompletableFuture<Map<String, InstanceRef>> getDartObjectProperties(
+      InspectorInstanceRef inspectorInstanceRef, final String[] propertyNames) {
+    return toObservatoryInstanceRef(inspectorInstanceRef).thenComposeAsync((InstanceRef instanceRef) -> {
+      final StringBuilder sb = new StringBuilder();
+      final List<String> propertyAccessors = new ArrayList<>();
+      final String objectName = "that";
+      for (String propertyName : propertyNames) {
+        propertyAccessors.add(objectName + "." + propertyName);
+      }
+      sb.append("<Object>[");
+      sb.append(Joiner.on(',').join(propertyAccessors));
+      sb.append("]");
+      final Map<String, String> scope = new HashMap<>();
+      scope.put(objectName, instanceRef.getId());
+      return getInstance(inspectorLibrary.eval(sb.toString(), scope)).thenApplyAsync(
```

Comment:
Ah, this is shorthand for individually eval'ing each getter?

Reply:
Exactly. Added that to the comment.